posts/2019/12/Spyder-terminal.md

```diff
@@ -0,0 +1,54 @@
+<!--
+.. title: Creating the ultimate terminal experience in Spyder 4 with spyder-terminal
+.. slug: creating-the-ultimate-terminal-experience-in-Spyder-4-with-spyder-terminal
+.. date: 2019-12-23 16:00:00 UTC-05:00
+.. author: Stephannie Jimenez
+.. tags: Labs, Spyder
+.. category:
+.. link:
+.. description:
+.. type: text
+-->
+
+The Spyder-terminal project is revitalized! In the new release of the project we were able to add numerous features that improve the terminal experience. Also, this new version enhances compatibility with the latest Spyder 4 release. We were able to develop all this work thanks to the latest release of the [xterm.js](https://xtermjs.org/) project.
+
+<!-- TEASER_END -->
+
+## ES6/JSX syntax
+In the first place, we were able to update all the old JavaScript files to a ES6/JS syntax. This change simplified the code base and maintenance for the terminal. Also, it allows us to extend the project in an easier way to new functionalities that the xterm.js API offers. In order to compile this code and run it inside Spyder we migrated our deployment to Webpack. Additionally, the tests for the client terminal were updated to support the new syntax.
+
+## Multiple shells per operative system
+
+A new feature added to the terminal is that the user can configure which shell to use in the terminal. Now the terminal is able to start with `bash`, `sh`, `ksh`, `zsh`, `csh`, `pwsh`, `tcsh`, `screen`, `tmux`, `dash` and `rbash` in UNIX systems and `cmd` and `powershell` in Windows. This option is given on the preferences pane and a restart of Spyder is required to apply the changes.
+
+![UNIX shell options for starting the terminal](/images/spyder-terminal/shells.png)
+
+This is a great feature because it allows the user to determine their shell interpreter among the ones that are installed in their systems. In this way, Spyder-terminal can be configured with any of the existing shells as long as it is available on their machine. Some examples of the configurable options of the interpreter are shown below, with bash and tcsh.
+
+![Spyder-terminal running on a bash shell](/images/spyder-terminal/bash.png)
+
+![Spyder-terminal running on a tcsh shell](/images/spyder-terminal/tcsh.png)
+
+## Theme configuration
+
+Another big change in this new version of the terminal is that now all the Spyder themes options are available. In this way, when a user changes the palette that wants to use with Spyder, automatically the terminal will use the same theme including the font size. Some examples of the new supported themes for the terminal are shown below.
+
+![Spyder-terminal with minimal color theme](/images/spyder-terminal/minimal.png)
```

Comment:
```suggestion
```

Its rather redundant to show both and it results in a double image block that should be avoided whenever practicable.